### PR TITLE
Add back the key not found diagnostic code

### PR DESCRIPTION
**Pull request by lasinicl**
## Purpose
$title

Fixes #<Issue Number>

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/ballerinalang/util/diagnostic/DiagnosticErrorCode.java [hunk 1]
```diff
@@ -512,7 +512,8 @@
     EXPRESSION_IS_NOT_A_CONSTANT_EXPRESSION("BCE3504", "expression.is.not.a.constant.expression"),
     INVALID_CONST_EXPRESSION("BCE3505", "invalid.const.expression"),
     CONSTANT_EXPRESSION_NOT_SUPPORTED("BCE3506", "const.expression.not.supported"),
-    CONSTANT_DECLARATION_NOT_YET_SUPPORTED("BCE3507", "constant.declaration.not.yet.supported.for.type"),
+    KEY_NOT_FOUND("BCE3507", "key.not.found"),
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/org/ballerinalang/util/diagnostic/DiagnosticErrorCode.java:515]: Where/how is this being used in stdlib? @lasinicl @niveathika

Comment (niveathika) [reply compiler/ballerina-lang/src/main/java/org/ballerinalang/util/diagnostic/DiagnosticErrorCode.java:515]: Used in the file compiler plugin https://github.com/ballerina-platform/module-ballerina-file/blob/master/file-compiler-plugin/src/main/java/io/ballerina/stdlib/file/compiler/FileServiceValidator.java#L96

Comment (MaryamZi) [reply compiler/ballerina-lang/src/main/java/org/ballerinalang/util/diagnostic/DiagnosticErrorCode.java:515]: I think we need to validate if this is the correct error code to use. That logic seems to be validating if the remote qualifier is present in a method.

Let's go with this change for now, but shall we create two issues for lang and the file module?

**Review by MaryamZi — APPROVED** [review]

File: compiler/ballerina-lang/src/main/java/org/ballerinalang/util/diagnostic/DiagnosticErrorCode.java (hunk 1 above)
Comment (MaryamZi) [reply compiler/ballerina-lang/src/main/java/org/ballerinalang/util/diagnostic/DiagnosticErrorCode.java:515]: Also, I was under the impression that there wouldn't/shouldn't be usages of these codes outside the compiler. 

Looping in @pubudu91 and @hasithaa  too.

Comment (lasinicl) [reply compiler/ballerina-lang/src/main/java/org/ballerinalang/util/diagnostic/DiagnosticErrorCode.java:515]: Created https://github.com/ballerina-platform/ballerina-lang/issues/30341

Comment (pubudu91) [reply compiler/ballerina-lang/src/main/java/org/ballerinalang/util/diagnostic/DiagnosticErrorCode.java:515]: The expectation was that compiler plugin developers would come up with their own diagnostics and diagnostic codes. I can't really think of a scenario where a compiler plugin would need a diagnostic from the compiler. 🤔 I mean, if that's the case, then that check could've been done in the compiler itself right? 🤔

**Comment by niveathika** [comment]
Closing this as we decided to use different error code in file for now

File: compiler/ballerina-lang/src/main/java/org/ballerinalang/util/diagnostic/DiagnosticErrorCode.java (hunk 1 above)
Comment (hasithaa) [reply compiler/ballerina-lang/src/main/java/org/ballerinalang/util/diagnostic/DiagnosticErrorCode.java:515]: As Pubudu mentioned, the expectation was each compiler plugin must have its own diagnostics and diagnostic codes. Diagnostics should be searchable using diagnostic codes, so It is wrong to use compiler codes.

**Pull request closed without merging**